### ISSUE TITLE
[Bug] Shortcode input errors require A1111 full restart

### ISSUE THREAD

**Issue by jhlchu** [ThereforeGames/unprompted#93]
**Describe the Problem**
When using zoom_enhance, error inputs (e.g. missing spaces like 0.3mask_size_max=0.3) causes unprompted to error out. Subsequent runs with corrected input fail to trigger shortcodes until A1111 is restart again.

**Expected Behavior**
After correcting error line, shortcodes should be able to run again without restart.

No errors are output to console when it fails to run again.

```
Traceback (most recent call last):
  File "P:\Computing\AI\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shortcodes.py", line 117, in render
    return str(self.handler(self.token.keyword, self.pargs, self.kwargs, context))
  File "P:\Computing\AI\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shared.py", line 61, in handler
    return(self.shortcode_objects[f"{keyword}"].run_atomic(pargs, kwargs, context))
  File "P:\Computing\AI\stable-diffusion-webui\extensions\unprompted/shortcodes\stable_diffusion\zoom_enhance.py", line 43, in run_atomic
    denoising_max = float(kwargs["denoising_max"]) if "denoising_max" in kwargs else 0.65
ValueError: could not convert string to float: '0.3mask_size_max=0.3'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "P:\Computing\AI\stable-diffusion-webui\modules\scripts.py", line 436, in postprocess
    script.postprocess(p, processed, *script_args)
  File "P:\Computing\AI\stable-diffusion-webui\extensions\unprompted\scripts\unprompted.py", line 455, in postprocess
    Unprompted.shortcode_objects[i].after(p,processed)
  File "P:\Computing\AI\stable-diffusion-webui\extensions\unprompted/shortcodes\basic\after.py", line 32, in after
    self.Unprompted.process_string(self.Unprompted.parse_alt_tags(content,"after"))
  File "P:\Computing\AI\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shared.py", line 157, in parse_alt_tags
    return(self.shortcode_parser.parse(string,context))
  File "P:\Computing\AI\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shortcodes.py", line 219, in parse
    return stack.pop().render(context)
  File "P:\Computing\AI\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shortcodes.py", line 58, in render
    return ''.join(child.render(context) for child in self.children)
  File "P:\Computing\AI\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shortcodes.py", line 58, in <genexpr>
    return ''.join(child.render(context) for child in self.children)
  File "P:\Computing\AI\stable-diffusion-webui\extensions\unprompted\lib_unprompted\shortcodes.py", line 121, in render
    raise ShortcodeRenderingError(msg) from ex
lib_unprompted.shortcodes.ShortcodeRenderingError: An exception was raised while rendering the 'zoom_enhance' shortcode in line 1.
```

**Comment by nseq** [comment]
I receive similar error whenever I interrupt image generation. After that the only way for Unprompted to start working again is to reload UI from extensions menu or to restart webui altogether.

**Comment by ThereforeGames** [comment]
Hi,

I have implemented a fix for this in v7.8.0. Please try updating the extension and let me know if the issue persists. Thank you!

**Comment by nseq** [comment]
Using latest version and still experiencing the same issue when I interrupt image generation, here's my error message, idk if it's much different than OP's:
```
Error running postprocess: /kaggle/working/stable-diffusion-webui/extensions/_unprompted/scripts/unprompted.py
Traceback (most recent call last):
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/lib_unprompted/shortcodes.py", line 117, in render
    return str(self.handler(self.token.keyword, self.pargs, self.kwargs, context))
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/lib_unprompted/shared.py", line 61, in handler
    return(self.shortcode_objects[f"{keyword}"].run_atomic(pargs, kwargs, context))
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/shortcodes/stable_diffusion/zoom_enhance.py", line 168, in run_atomic
    fixed_image = self.Unprompted.shortcode_objects["img2img"].run_atomic(set_pargs,None,None)
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/shortcodes/stable_diffusion/img2img.py", line 87, in run_atomic
    return img2img_images[0]
IndexError: list index out of range

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/kaggle/working/stable-diffusion-webui/modules/scripts.py", line 436, in postprocess
    script.postprocess(p, processed, *script_args)
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/scripts/unprompted.py", line 455, in postprocess
    Unprompted.shortcode_objects[i].after(p,processed)
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/shortcodes/basic/after.py", line 32, in after
    self.Unprompted.process_string(self.Unprompted.parse_alt_tags(content,"after"))
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/lib_unprompted/shared.py", line 157, in parse_alt_tags
    return(self.shortcode_parser.parse(string,context))
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/lib_unprompted/shortcodes.py", line 219, in parse
    return stack.pop().render(context)
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/lib_unprompted/shortcodes.py", line 58, in render
    return ''.join(child.render(context) for child in self.children)
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/lib_unprompted/shortcodes.py", line 58, in <genexpr>
    return ''.join(child.render(context) for child in self.children)
  File "/kaggle/working/stable-diffusion-webui/extensions/_unprompted/lib_unprompted/shortcodes.py", line 121, in render
    raise ShortcodeRenderingError(msg) from ex
lib_unprompted.shortcodes.ShortcodeRenderingError: An exception was raised while rendering the 'zoom_enhance' shortcode in line 1.
```
If you want to reproduce this error just start generating an image and manually interrupt generation, after that Unprompted stops working until you reload UI from extenstions menu.

**Comment by ThereforeGames** [comment]
Hi @nseq,

I think I managed to fix this in v7.9.0 (for real this time) - could you give it another try and let me know how it goes?

Thank you!

**Comment by ThereforeGames** [comment]
Closing due to lack of activity. Please let me know if this issue persists.
